starcheck/calc_ccd_temps.py

```diff
@@ -195,10 +201,24 @@ def get_interval_temps(intervals, times, ccd_temp):
         stop_idx = 1 + np.searchsorted(times, interval['tstop'])
         start_idx = -1 + np.searchsorted(times, interval['tstart'])
         ok_temps = ccd_temp[start_idx:stop_idx]
+        ok_times = times[start_idx:stop_idx]
         if len(ok_temps) > 0:
             obs['ccd_temp'] = np.max(ok_temps)
             obs['n100_warm_frac'] = dark_model.get_warm_fracs(
                 100, interval['tstart'], np.max(ok_temps))
+        if obsreqs is None:
+            obsreqs = {}
+        if interval['obsid'] in obsreqs and len(ok_temps) > 0:
```

Comment:
You can skip lines 209 and 210 and do `if obsreqs and interval['obsid'] ..`.  Or at least move 209,210 outside the loop.